js/ajax.js

```diff
@@ -22,6 +22,11 @@ $(document).ready(function() {
 		event.preventDefault();
 		OC.msg.startSaving('#password_policy .msg');
 
+		if (($("input[name=spv_def_special_chars_checked]").is(':checked') === true) &&
+			($("input[name=spv_def_special_chars_value]").val() === "")) {
+			OC.Notification.showTemporary(t('password_policy', 'Error: The special characters cannot be empty.'));
```

Comment:
````suggestion
			OC.Notification.showTemporary(t('password_policy', 'Error: The special characters cannot be empty.'));
 			OC.msg.finishedSaving('#password_policy .msg', {status: "failed", data: { message: t('password_policy', 'Failed to save!')} });
 			return;
````
If we can detect an error in client side, I guess no need to post erroneous request to server.  
Also, currently, we are saving `spv_def_special_chars_checked` config value even we say `Failed to save`. Suggested change prevents this situation also .

Reply:
I was trying to be in the defensive side. In either way the data shouldn't be saved, be it from the client side post or any hackish thing which reaches the server.

Reply:
Made changes as discussed.